app/services/calculators/property_calculator.rb

```diff
@@ -1,43 +1,66 @@
 module Calculators
   class PropertyCalculator
-    attr_writer :remaining_mortgage_allowance
+    Result = Struct.new(:transaction_allowance,
+                        :net_value,
+                        :net_equity,
+                        :main_home_equity_disregard,
+                        :assessed_equity,
+                        :smod_applied,
+                        keyword_init: true)
 
     class << self
-      def call(submission_date:, capital_summary:, level_of_representation:)
-        new(submission_date:, capital_summary:, level_of_representation:).call
-      end
-    end
+      def call(submission_date:, properties:, level_of_representation:, smod_level:)
+        remaining_mortgage_allowance ||= Threshold.value_for(:property_maximum_mortgage_allowance, at: submission_date)
 
-    def initialize(submission_date:, capital_summary:, level_of_representation:)
-      @submission_date = submission_date
-      @capital_summary = capital_summary
-      @level_of_representation = level_of_representation
-    end
+        Property.transaction do
+          (properties.select(&:main_home) + properties.reject(&:main_home)).map do |property|
+            allowable_outstanding_mortgage = calculate_outstanding_mortgage(property, remaining_mortgage_allowance)
+            remaining_mortgage_allowance -= allowable_outstanding_mortgage
+            assessor_result = Assessors::PropertyAssessor.call(property:,
```

Comment:
PropertyAssessor is no longer in charge of coming up with a figure for `assessed_equity`. It's sort of doing half the thinking. So I think either should it be renamed, because it's not really assessing the property, or should the property assessment logic stay inside PropertyAssessor, and it could receive `smod_remaining` as an argument and return `smod_applied` as part of its result struct so that PropertyCalculator can track how much is left in the SMOD allowance?

Reply:
That sounds reasonable, since at the moment the division of responsibility is a little makeshift.